model/patch_lifecycle.go

```diff
@@ -506,7 +525,9 @@ func FinalizePatch(ctx context.Context, p *patch.Patch, requester string, github
 				p.Githash)
 		}
 	}
-	intermediateProject.Id = p.Id.Hex()
+	if mustInsertParserProjectDuringFinalization {
```

Comment:
Can this be moved into the block on L703? `if mustInsertParserProjectDuringFinalization {` is used multiple times in this function and it seems they can be consolidated so it's more clear what the function of the flag is.

Reply:
Done.